src/Cake.Common/Tools/DupFinder/DupFinderRunner.cs

```diff
@@ -0,0 +1,212 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using Cake.Core;
+using Cake.Core.IO;
+using Cake.Core.Utilities;
+
+namespace Cake.Common.Tools.DupFinder
+{
+    /// <summary>
+    /// DupFinder runner
+    /// </summary>
+    public sealed class DupFinderRunner : Tool<DupFinderSettings>
+    {
+        private readonly ICakeEnvironment _environment;
+        private readonly IGlobber _globber;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DupFinderRunner"/> class.
+        /// </summary>
+        /// <param name="fileSystem">The file system.</param>
+        /// <param name="environment">The environment.</param>
+        /// <param name="processRunner">The process runner.</param>
+        /// <param name="globber">The globber</param>
+        public DupFinderRunner(IFileSystem fileSystem, ICakeEnvironment environment, IProcessRunner processRunner, IGlobber globber)
+            : base(fileSystem, environment, processRunner, globber)
+        {
+            _environment = environment;
+            _globber = globber;
+        }
+
+        /// <summary>
+        /// Analyses the specified files using the specified settings.
+        /// </summary>
+        /// <param name="files">The files.</param>
+        /// <param name="settings">The settings.</param>
+        public void Run(IEnumerable<FilePath> files, DupFinderSettings settings)
+        {
+            if (settings == null)
+            {
+                throw new ArgumentNullException("settings");
+            }
+
+            if (files == null)
+            {
+                throw new ArgumentNullException("files");
+            }
+
+            Run(settings, GetArgument(settings, files));
+        }
+
+        /// <summary>
+        /// Analyses all files matching the specified pattern using the specified settings.
+        /// </summary>
+        /// <param name="pattern">The pattern.</param>
+        /// <param name="settings">The settings.</param>
+        public void Run(string pattern, DupFinderSettings settings)
+        {
+            if (pattern == null)
+            {
+                throw new ArgumentNullException("pattern");
+            }
+            var sourceFiles = _globber.Match(pattern).OfType<FilePath>();
+
+            Run(sourceFiles, settings);
+        }
+
+        /// <summary>
+        /// Runs ReSharper's DupFinder using the provided config file.
+        /// </summary>
+        /// <param name="configFile">The config file.</param>
+        public void RunFromConfig(FilePath configFile)
```

Comment:
I think this method could be moved to the alias. This way we would have only one entry method to the tool.